CMakeLists_files.cmake

```diff
@@ -32,6 +32,11 @@ macro (ADD_CUDA_OR_HIP_FILE LIST DIR FILE)
 
   if(CUDA_FOUND AND NOT CONVERT_CUDA_TO_HIP)
     list (APPEND ${LIST} "${DIR}/gpuistl/${FILE}")
+
+    # .cu files might struggle with evaluating things at compile-time, this flag solves those issues
+    if(${FILE} MATCHES "\\.cu$")
+      set_source_files_properties("${DIR}/gpuistl/${FILE}" PROPERTIES COMPILE_FLAGS "--expt-relaxed-constexpr")
```

Comment:
This flag is the simplest way to allow cuda to compile code where std::arrays are used in code that will be run on the GPU, because for some reason constexpr indexing is not allowed otherwise... Is this the proper way to add compilation flags in OPM, and does this look okay @akva2

Reply:
yes this is okay, although i'd like to see a more focused approach, ie, a list with files where it is needed rather than bulk for every .cu file (unless it is required for all but i don't think it it).

Reply:
I can add a boolean argument to the function such that the ones needing this flag can pass an extra true to it while the others use false, it will clutter the code a bit since there are many calls to the macro though...

Reply:
you can set source file properties whenever. i don't see why this can't be
```cmake
set(LIST_OF_FILES foo1.cu foo2.cu)
if(NOT CONVERT_CUDA_TO_HIP)
  foreach(file ${LIST_OF_FILES})
    set_source_file_properties(${file} blablabla)
  endforeach()
endif()
```
?

Reply:
the annoyance is actually a big part of my reasoning for not doing that but okay, i won't insist.

Reply:
I trust your judgement and have rewritten it to list the problematic .cu files and add the property explicitly without change the existing hipification macro